src/checks/substitution_key.rs

```diff
@@ -0,0 +1,177 @@
+use crate::checks::Check;
+use crate::common::*;
+
+pub(crate) struct SubstitutionKeyChecker<'a> {
+    name: &'a str,
+    template: &'a str,
+}
+
+impl Default for SubstitutionKeyChecker<'_> {
+    fn default() -> Self {
+        Self {
+            name: "SubstitutionKey",
+            template: "The {} key is not assigned properly",
+        }
+    }
+}
+
+impl Check for SubstitutionKeyChecker<'_> {
+    fn run(&mut self, line: &LineEntry) -> Option<Warning> {
+        let mut value = match line.get_value().map(str::trim) {
+            Some(value) if !value.starts_with('\'') => value,
+            _ => return None,
+        };
+
+        let is_escaped =
+            |prefix: &str| prefix.chars().rev().take_while(|ch| *ch == '\\').count() % 2 == 1;
+
+        // Checks if keys used in value have both '{' '}' or neither
+        while let Some(index) = value.find('$') {
+            let prefix = &value[..index];
+            let raw_key = &value[index + 1..];
+
+            // Separate initial key from the rest
+            let (initial_key, rest) = raw_key
+                .find(|c: char| c == '$')
```

Comment:
Here you can simplify the predicate
```suggestion
                .find('$')
```